### PR TITLE
[DeadCode] Handle file_get_contents() only remove variable on RemoveUnusedVariableAssignRector

### LINKED ISSUE #8263: RemoveUnusedVariableAssignRector: calls with side-effects should not be removed
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/a56b4a20-48c9-4a49-8316-ba6873ba3c76

```php
<?php

final class DemoFile
{
    public function run(bool $param)
    {
        $return = file_get_contents("http://example.com");
    }
}
```

### Responsible rules

RemoveUnusedVariableAssignRector

## Expected Behavior

calling `file_get_contents` has side-effects and therefore this line should not be removed, but just the assigned variable (or just leave the line as is)

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8263

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @staabm

File: rules/DeadCode/SideEffect/PureFunctionDetector.php [hunk 1]
```diff
@@ -33,6 +33,8 @@ final class PureFunctionDetector
         // http
         'header', 'header_remove', 'http_response_code', 'setcookie',
 
+        'file_get_contents',
```
Comment (staabm) [inline rules/DeadCode/SideEffect/PureFunctionDetector.php:36]: ohh nice, didn't realize we have such a list. PHPStan also has such a list.. maybe we can utilize the `hasSideEffects` methods of the different Reflection*-classes?

https://github.com/search?q=repo%3Aphpstan%2Fphpstan-src%20hasSideEffects&type=code

if we would trust PHPStans list, we would not need to duplicate that list in rector itself

Comment (samsonasik) [reply rules/DeadCode/SideEffect/PureFunctionDetector.php:36]: no, phpstan has very limited list, while psalm has more list, so it use list form psalm, if we can have script to make both tools collect more list on each other, that would be awesome :)

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Pull request merged**